### PR TITLE
59401274 rebase and fix pull request  168

### PR DESCRIPTION
**Pull request by ke4**

### CODE REVIEW

File: lib/lims-api/context.rb [hunk 1]
```diff
@@ -312,6 +318,38 @@ def recursively_load_uuid(attributes, session)
...
+      #--------------------------------------------------
+        # Message Bus
+      #--------------------------------------------------
+
+      # Publish a message on the bus and route it with a routing key.
+      # @param [Class, String] action 
+      # @param [Hash, nil] resource to publish 
+      def publish(action, resource)
+        action = find_action_name(action) unless action.is_a? String
+        user ||= "user"
```
Comment (maxigit) [inline lib/lims-api/context.rb]: I don't think that's needed. (Or at least don't modify @user)

File: lib/lims-api/context.rb [hunk 2]
```diff
@@ -312,6 +318,38 @@ def recursively_load_uuid(attributes, session)
...
+        payload = message_payload(action, user, resource)
```
Comment (maxigit) [inline lib/lims-api/context.rb]: do we need user as a parameter ?

File: lib/lims-api/context.rb [hunk 3]
```diff
@@ -234,7 +240,7 @@ def execute_action(action)
         # @todo add user and 
         def create_action(action_class, attributes, resource_class=nil)
           resource_class ||= resource_class_for_class(action_class)
-          action = action_class.new( :store => store, :user => "user", :application => "application") do |a, session|
+          action = action_class.new( :store => store, :user => user, :application => application_id) do |a, session|
```
Comment (maxigit) [inline lib/lims-api/context.rb]: What is the default value for application_id ?
I think application_id and pipeline_id are the same, aren't they ?

Comment (ke4) [inline lib/lims-api/context.rb]: Application_id is coming form ContextService and its default value is nil.
In laboratory_app we need to change one of the spec_helper to set it when we create the ContextService.

Application_id and pipeline_id is not the same.
Application_id for example "laboratory_app", pipeline_id for example "Illumina A".

On 25 Oct 2013, at 14:56, maxigit wrote:

> In lib/lims-api/context.rb:
> 
> > @@ -234,7 +240,7 @@ def execute_action(action)
> >          # @todo add user and 
> >          def create_action(action_class, attributes, resource_class=nil)
> >            resource_class ||= resource_class_for_class(action_class)
> > -          action = action_class.new( :store => store, :user => "user", :application => "application") do |a, session|
> > -          action = action_class.new( :store => store, :user => user, :application => application_id) do |a, session|
> >   What is the default value for application_id ?
> >   I think application_id and pipeline_id are the same, aren't they ?
> 
> —
> Reply to this email directly or view it on GitHub.

## 

 The Wellcome Trust Sanger Institute is operated by Genome Research 
 Limited, a charity registered in England with number 1021457 and a 
 company registered in England with number 2742969, whose registered 
 office is 215 Euston Road, London, NW1 2BE.

**Comment by maxigit** [comment]
Merged in pull request #192.

**Pull request closed without merging**